DeliveriesEngine/src/main/java/pt/deliveries/deliveries_engine/controller/VehicleController.java

```diff
@@ -0,0 +1,33 @@
+package pt.deliveries.deliveries_engine.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import pt.deliveries.deliveries_engine.Model.Vehicle;
+import pt.deliveries.deliveries_engine.Service.VehicleServiceImpl;
+
+import java.util.logging.Logger;
+
+@RestController
+@RequestMapping("/deliveries-api/vehicle")
+public class VehicleController {
+
+    @Autowired
+    private VehicleServiceImpl service;
+
+    private Logger logger = Logger.getLogger(CourierController.class.getName());
+
+    @RequestMapping("/create")
+    public ResponseEntity<Vehicle> createVehicle(@RequestBody Vehicle vehicle){
```

Comment:
Persistent entities should not be used as arguments of "@RequestMapping" methods

Replace this persistent entity with a simple POJO object